WebRoot/js/zimbraMail/share/view/dialog/ZmTwoFactorSetupDialog.js

```diff
@@ -287,53 +579,51 @@ function(currentDivId, result) {
 		if (enableTwoFactorAuthResponse.csrfToken && enableTwoFactorAuthResponse.csrfToken[0] && enableTwoFactorAuthResponse.csrfToken[0]._content) {
 			window.csrfToken = enableTwoFactorAuthResponse.csrfToken[0]._content;
 		}
-		var secret = enableTwoFactorAuthResponse.secret;
-		var scratchCodes = enableTwoFactorAuthResponse.scratchCodes;
-		if (secret && secret[0] && secret[0]._content) {
-			Dwt.setInnerHtml(this._keySpan, secret[0]._content);
-			this._handleTwoFactorAuthSuccess(currentDivId);
-			return;
-		}
-		else if (scratchCodes && scratchCodes[0] && scratchCodes[0].scratchCode) {
-			if (typeof appCtxt !== "undefined") {
-				//Only the server will set ZmSetting.TWO_FACTOR_AUTH_ENABLED. Don’t try to save the setting from the UI.
-				appCtxt.set(ZmSetting.TWO_FACTOR_AUTH_ENABLED, true, false, false, true);
+
+		if (typeof appCtxt !== "undefined") {
+			// scratchCodes is returned only when a first method is enabled.
```

Comment:
The comment is wrong. Scratch codes are returned when a 2nd method is enabled as well. They are the existing codes.
i.e. they are the same as the ones generated at a 1st method is enabled. If some of them have already been used, the remaining codes are returned.
I will remove the comment.

Reply:
done